modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/typeclasses/UnsafeRun.kt

```diff
@@ -0,0 +1,11 @@
+package arrow.effects.typeclasses
+
+import arrow.Kind
+import arrow.core.Either
+import arrow.effects.unsafe
+
+interface UnsafeRun<F> {
+  suspend fun <A> unsafe.runBlocking(fa: () -> Kind<F, A>): A
+  suspend fun <A> unsafe.runNonBlocking(fa: () -> Kind<F, A>, cb: (Either<Throwable, A>) -> Unit): Unit
```

Comment:
We call this `runAsync` even if it's not multithreaded.

Reply:
yes, and it's been changed to make sure of 2 added compile time guarantees:
1. This removes the need to have unsafe functions in the environment which means we can deprecate `unsafeRunSync` etc.
2. Calling `runBlocking` and `runNonBlocking` does not compile unless the user has explicitly imported the `unsafe` function which is the only function in `Arrow` that will be allowed to run Fx programs.

The names have been changed because these are more familiar idioms to Kotlin users and we have extra compile guarantees they are well aware that effects are performed at that point.